src/main/java/persistence/sql/ddl/metadata/ColumnMetadata.java

```diff
@@ -0,0 +1,47 @@
+package persistence.sql.ddl.metadata;
+
+import jakarta.persistence.Transient;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ColumnMetadata {
+    private final List<Column> columns;
+
+    private ColumnMetadata(List<Column> columns) {
+        validate(columns);
+        this.columns = columns;
+    }
+
+    public static ColumnMetadata from(Class<?> clazz) {
+        return Arrays.stream(clazz.getDeclaredFields())
+                .filter(ColumnMetadata::isNotTransient)
+                .map(Column::from)
+                .collect(Collectors.collectingAndThen(Collectors.toList(), ColumnMetadata::new));
+    }
+
+    private static boolean isNotTransient(Field field) {
+        return !field.isAnnotationPresent(Transient.class);
+    }
+
+    private void validate(List<Column> columns) {
+        boolean hasIdAnnotation = columns.stream()
+                .anyMatch(Column::isPrimaryKey);
+
+        if (!hasIdAnnotation) {
+            throw new IllegalArgumentException("@Id가 필수로 지정되어야 합니다");
+        }
+    }
+
+    public List<Column> getColumns() {
+        return columns;
+    }
```

Comment:
Column 들을 가지고 있는 `ColumnMetadata` 를 추출해주셨네요 👍

그런데 단순히 가진 필드를 getter 로 꺼내는건 조금은 아쉽네요. 해당 클래스에 좀 더 책임을 부여하거나, 최소한 외부에서 list 를 수정하지 못하도록 불변하게 만들어보는건 어떨까요?
```suggestion
    public List<Column> getColumns() {
        return Collections.unmodifiableList(columns);
    }
```